src/decompiler.cc

```diff
@@ -28,6 +28,105 @@
 
 namespace wabt {
 
+template<typename T>
+void RenameToIdentifiers(std::vector<T*>& things, BindingHash& bh,
+                         const std::set<string_view>* filter) {
+  Index i = 0;
+  for (auto thing : things) {
+    // Filter out non-identifier characters, and try to reduce the size of
+    // gigantic C++ signature names.
+    std::string s;
+    size_t nesting = 0;
+    size_t read = 0;
+    size_t word_start = 0;
+    for (auto c : thing->name) {
```

Comment:
To reduce code bloat? sure..

Reply:
Actually going to move all of this to its own file while I am at it.

Reply:
And to reduce nesting and increase readability and re-useability. 

I guess I'm also a fan or using template minimally I and realized the bulk of this function didn't depend on the template param so it could be refactored.   So I'm showing my biases here maybe...

Reply:
It reduced 1 nesting level.
It is less readable to me. I prefer code to be as flat as possible.
Re-useability would not be a good reason, seeing as there is only 1 caller. We have no idea what a second caller would need, if there ever is one.

Reply:
I agree re-usability is a weak argument in this case.  I take that one back :)  But naming the sub-routine for readability and reducing nesting by 1 level are both worth doing in my book.

I see this version as flatter since it has less syntactic nesting.   I guess its personal preference I'm not going to push hard for these kind of changes in general.   I'm personally always tempted to extract a named function when I see a long function with two or more levels of testing that take up most of the body.   Some of that ruby on rails must of rubbed off on me at some point I guess.